gmso/utils/connectivity.py

```diff
@@ -120,14 +114,12 @@ def _detect_connections(compound_line_graph, top, type_="angle"):
     sorted_conn_matches = list()
     for match in conn_matches:
         if type_ in ("angle", "dihedral"):
-            if top.get_index(match[0]) < top.get_index(match[-1]):
+            if match[0] < match[-1]:
                 sorted_conn = match
             else:
                 sorted_conn = match[::-1]
         elif type_ == "improper":
-            latter_sites = sorted(
-                match[1:], key=lambda site: top.get_index(site)
-            )
+            latter_sites = sorted(match[1:])
```

Comment:
This can be condensed into sorted_conn = [match[0]] + sorted(match]
```suggestion
            sorted_conn = [match[0]] + sorted(match[1:])
```

Reply:
added in latest commit